integration/entity.go

```diff
@@ -0,0 +1,92 @@
+package integration
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/metric"
+	"github.com/newrelic/infra-integrations-sdk/persist"
+	"github.com/pkg/errors"
+)
+
+// Entity is the producer of the data. Entity could be a host, a container, a pod, or whatever unit of meaning.
+type Entity struct {
+	lock      sync.Mutex
+	Metadata  EntityMetadata   `json:"entity"`
+	Metrics   []*metric.Set    `json:"metrics"`
+	Inventory metric.Inventory `json:"inventory"`
+	Events    []*metric.Event  `json:"events"`
+	storer    persist.Storer
+}
+
+// EntityMetadata stores entity Metadata
+type EntityMetadata struct {
+	Name string `json:"name"`
+	Type string `json:"type""`
+}
+
+// EntityID entity identifier
+type EntityID string
+
+// NewEntity creates a new remote-entity.
+func NewEntity(entityName, entityType string, storer persist.Storer) (*Entity, error) {
```

Comment:
I want to raise 2 concerns here. I will write them in 2 separate comments

Reply:
I totally agree, but this involves an unplanned refactor, we may add it for the next iteration. Is that ok?

Reply:
Sure!